src/components/media/UserGallery.tsx

```diff
@@ -120,6 +121,34 @@ export default function UserGallery ({ uid, postId: initialPostId, auth, userPro
     setSlideNumber(-1)
   }, [])
 
+  const [imageListToShow, setImageListToShow] = useState<MediaType[]>([])
+  const [imageListToShowLength, setImageListToShowLength] = useState(20)
+
+  useEffect(() => {
+    // add new items from the imageList to the imageListToShow array when length changes
+    if (imageList !== undefined) setImageListToShow(imageList.slice(0, imageListToShowLength))
+    console.log('Length Before ', imageListToShow.length)
+  }, [imageList, imageListToShowLength])
+
+  useEffect(() => {
+    console.log('Length After ', imageListToShowLength)
+
+    const handleScroll = (): void => {
```

Comment:
Sorry I forgot to mention in our chat on Discord that we're current using `react-use` library that provides a wealth of useful hooks.   You can use the `useIntersection()` hook (built on Intersection Observer API) to simplify the scroll detection code.
- Docs: https://streamich.github.io/react-use/?path=/story/sensors-useintersection--docs
- StickyHeader component https://github.com/OpenBeta/open-tacos/blob/develop/src/components/crag/StickyHeader.tsx

In StickyHeader variable `atTop` becomes true when the `<div ref={intersectionRef} ...>` reaches the top of the screen.   You can change `rootMargin` to be notified when you reach the bottom of the screen (I don't remember exactly how on top of my head).

Reply:
I tried this but not working, maybe I'll have to dive deeper into the working of useIntersection() hook (learning react).
But, I solved the problem of custom code. I implemented this again using react-infinite-scroll-component library. It was quite simple to use that which I didn't expect.